File: courses/machine_learning/section_6/Simple Linear Regression/Python/simple_linear_regression.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# Import data
dataset = pd.read_csv("Salary_Data.csv")
# matrix of features or independent values
X = dataset.iloc[:, :-1].values
# matrix of dependent variable vector
y = dataset.iloc[:, -1].values

# Split data to Test set and training set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

# Training simple linear regression on the training set
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting the Test results
y_pred = regressor.predict(X_test)
# print(y_pred)

# Get single prediction Ex. from Person who has 12 years of experience.
print(regressor.predict([[12]]))

# Visualising Training set results
plt.scatter(X_train, y_train, color="red")
plt.plot(X_train, regressor.predict(X_train), color="blue")
plt.title("Years of Experience vs Salary (Training set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

# Visualising Test set results
plt.scatter(X_test, y_test, color="red")
# No need to change these. The result will be the same regardless because of the unique equation, and
# we would end up whit same regression line.
plt.plot(X_train, regressor.predict(X_train), color="blue")
plt.title("Years of Experience vs Salary (Test set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

File: courses/machine_learning/section_6/Simple Linear Regression/R/simple_linear_regression.R
# Simple linear regression
dataset = read.csv("Salary_Data.csv")

# Splitting data into test set and training set
#install.packages("caTools")
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Fitting simple linear regression to the Training set
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

# Predicting the test set results
y_pred = predict(regressor, newdata = test_set)

# Visualising the Training set
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = "red") +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = "blue") +
  ggtitle("Years Of Experience vs Salary (Training set)") +
  xlab("Years of Experience") +
  ylab("Salary")

# Visualising the Test set
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = "red") +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = "blue") +
  ggtitle("Years Of Experience vs Salary (Test set)") +
  xlab("Years of Experience") +
  ylab("Salary")